pkg/operation/botanist/backupentry.go

```diff
@@ -90,3 +92,14 @@ func (b *Botanist) DeploySourceBackupEntry(ctx context.Context) error {
 	b.Shoot.Components.SourceBackupEntry.SetBucketName(bucketName)
 	return b.Shoot.Components.SourceBackupEntry.Deploy(ctx)
 }
+
+// DestroySourceBackupEntry destroys the source BackupEntry. It returns nil if the CopyEtcdBackupsDuringControlPlaneMigration feature gate
+// is disabled or the Seed backup is not enabled or the Shoot is in restore phase.
+func (b *Botanist) DestroySourceBackupEntry(ctx context.Context) error {
+	if !gardenletfeatures.FeatureGate.Enabled(features.CopyEtcdBackupsDuringControlPlaneMigration) ||
+		b.Seed.GetInfo().Spec.Backup == nil || !b.isRestorePhase() {
+		return nil
+	}
```

Comment:
Shouldn't we have this check also in `DeploySourceBackupEntry`? I know that it works without it, but only because it's checked by `IsCopyOfBackupsRequired` in the reconcile flow. Or, alternatively, could we encapsulate this check in a "predicate" function similar to `IsCopyOfBackupsRequired` to be similarly checked in the reconcile flow? The current implementation works, but is "asymmetrical" and this could cause some confusion.

Reply:
The recommendation from @plkokanov was not to reuse the existing `isCopyOfBackupsRequired` but rather to implement similar check inline as in this PR. @plkokanov feel free to add more details.

Reply:
`isCopyOfBackupsRequired` definitely should not be reused (copy of backups may not be required but the source backup entry may still exist and need to the deleted), I had something else in mind. Anyway, I don't insist, we should probably merge as it is once it's sufficiently tested.